kratos/mpi/python_scripts/distributed_import_model_part_utility.py

```diff
@@ -68,7 +69,10 @@ def ImportModelPart(self):
                 if not partition_in_memory:
                     ## Serial partition of the original .mdpa file
                     if self.comm.Rank() == 0:
-                        partitioner = KratosMetis.MetisDivideHeterogeneousInputProcess(model_part_io, number_of_partitions , domain_size, verbosity, sync_conditions)
+                        if model_part_import_settings["partition_sub_model_parts_list"].GetBool():
+                            partitioner = KratosMetis.MetisDivideSubModelPartsHeterogeneousInputProcess(model_part_io, model_part_import_settings, number_of_partitions , domain_size, verbosity, sync_conditions)
+                        else:
+                            partitioner = KratosMetis.MetisDivideHeterogeneousInputProcess(model_part_io, number_of_partitions , domain_size, verbosity, sync_conditions)
```

Comment:
so this only works when not partitioning in memory?

Reply:
honestly, I have not tried yet. It might be possible it works without much effort.